src/Tribe/Views/V2/Widgets/Widget_Abstract.php

```diff
@@ -289,4 +350,78 @@ protected function args_to_context( array $arguments, Context $context ) {
 
 		return $context_args;
 	}
+
+	/**
+	 * Handles gathering the data for admin fields.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                 $field_name    The ID of the field.
+	 * @param array<string,mixed> $field       The field info.
+	 * @param Context             $context     The Admin View current context.
+	 *
+	 * @return array<string,mixed> $data The assembled field data.
+	 */
+	public function get_admin_data( $arguments, $field_name, $field ) {
+		$data = [
+			'classes'     => Arr::get( $field, 'classes', '' ),
+			'dependency'  => $this->format_dependency( $field ),
+			'id'          => $this->get_field_id( $field_name ),
+			'label'       => Arr::get( $field, 'label', '' ),
+			'name'        => $this->get_field_name( $field_name ),
+			'options'     => Arr::get( $field, 'options', [] ),
+			'placeholder' => Arr::get( $field, 'placeholder', '' ),
+			'value'       => Arr::get( $arguments, $field_name ),
+		];
+
+		$children = Arr::get( $field, 'children', [] );
+
+		if ( ! empty( $children ) ) {
+			foreach ( $children as $child_name => $child ) {
+				$child_data = $this->get_admin_data(
+					$arguments,
+					$child_name,
+					$child
+				);
+
+				$data['children'][ $child_name ] = $child_data;
+			}
+		}
+
+		$data = array_merge( $field, $data );
+
+		return apply_filters( 'tribe_events_views_v2_widget_field_data', $data, $field_name, $this );
+
+		return $data;
```

Comment:
:warning: Code after the RETURN statement on line 393 cannot be executed

<sub>hash: 8a39ce522427220bcd10e79b3307c14b</sub>